clients/python-wrapper/lakefs/object.py

```diff
@@ -202,19 +202,15 @@ def read(self, read_bytes: int = None) -> TextIO | BinaryIO:
         if read_bytes and read_bytes <= 0:
             raise OSError("read_bytes must be a positive integer")
 
-        stat = self._obj.stat()
-        if self._pos >= stat.size_bytes:
-            raise EOFError
-        read_bytes = read_bytes if read_bytes is not None else stat.size_bytes
-        new_pos = min(self._pos + read_bytes, stat.size_bytes)
-        read_range = _RANGE_STR_TMPL.format(start=self._pos, end=new_pos - 1)
+        end_pos = self._pos + read_bytes - 1 if read_bytes is not None else ""
+        read_range = _RANGE_STR_TMPL.format(start=self._pos, end=end_pos)
         with api_exception_handler(_io_exception_handler):
             contents = self._client.sdk_client.objects_api.get_object(self._obj.repo,
                                                                       self._obj.ref,
                                                                       self._obj.path,
                                                                       range=read_range,
                                                                       presign=self._pre_sign)
-        self._pos = new_pos  # Update pointer position
+        self._pos += len(contents)  # Update pointer position
```

Comment:
do we need to add the content len after the decode?

Reply:
contents returned in bytes so positions should advance by len(contents)

Reply:
Right - my bad - this is return.